examples/conditional-request-handling/JsonRequestHandler.cpp

```diff
@@ -2,29 +2,27 @@
 
 #include "JsonRequestHandler.h"
 #include <string>
-#include "ConditionalRequestSubscription.h"
+#include "yarpl/v/Flowable.h"
+#include "yarpl/v/Flowables.h"
 
-using namespace ::reactivesocket;
+using namespace reactivesocket;
+using namespace rsocket;
+using namespace yarpl;
 
 /// Handles a new inbound Stream requested by the other end.
-void JsonRequestHandler::handleRequestStream(
-    Payload request,
-    StreamId streamId,
-    const std::shared_ptr<Subscriber<Payload>>& response) noexcept {
+yarpl::Reference<yarpl::Flowable<reactivesocket::Payload>>
+JsonRequestHandler::handleRequestStream(Payload request, StreamId streamId) {
   LOG(INFO) << "JsonRequestHandler.handleRequestStream " << request;
 
   // string from payload data
-  auto pds = request.moveDataToString();
-  auto requestString = std::string(pds, request.data->length());
+  const char* p = reinterpret_cast<const char*>(request.data->data());
```

Comment:
nit: I guess you got tired of writing this code so you eventually created Payload::cloneDataToString method.
Do you want to use it over here?